libs/platform/src/lib/components/panel/panel.component.ts

```diff
@@ -0,0 +1,151 @@
+import {
+    Component,
+    Input,
+    ChangeDetectorRef,
+    Output,
+    EventEmitter,
+    ContentChild,
+    ChangeDetectionStrategy,
+    ViewChild,
+    OnInit,
+    OnChanges,
+    SimpleChanges
+} from '@angular/core';
+
+import { PanelTitleDirective } from '@fundamental-ngx/core';
+
+import { BaseComponent } from '../base';
+import { ContentDensity } from '../form/form-control';
+
+import { PanelConfig } from './panel.config';
+import { PanelActionsComponent } from './panel-actions/panel-actions.component';
+import { PanelContentComponent } from './panel-content/panel-content.component';
+
+/** Panel change event instance */
+export class PanelExpandChangeEvent {
+    constructor(public source: PanelComponent, public payload: boolean) {}
+}
+
+/**
+ * Fundamental Panel component
+ *
+ * ```html
+ * <fdp-panel title="Panel Header">
+ *     <fdp-panel-content>
+ *        Panel Content
+ *     </fdp-panel-content>
+ * </fdp-panel>
+ * ```
+ *
+ * */
+@Component({
+    selector: 'fdp-panel',
+    templateUrl: './panel.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PanelComponent extends BaseComponent implements OnInit, OnChanges {
+    /**
+     * sets Panel title.
+     */
+    @Input()
+    title: string;
+
+    /**
+     * Whether the Panel Content is expanded
+     */
+    @Input()
+    expanded = true;
+
+    /**
+     * Whether the Panel is expandable
+     */
+    @Input()
+    expandable = true;
+
+    /**
+     * content Density of element. 'cozy' | 'compact'
+     */
+    @Input()
+    set contentDensity(contentDensity: ContentDensity) {
+        this._contentDensity = contentDensity;
+        this.isCompact = contentDensity === 'compact';
+    }
+
+    /**
+     * ARIA label for button when the Panel is collapsed
+     */
+    @Input()
+    expandLabel = this._panelConfig.expandLabel;
+
+    /**
+     * ARIA label for button when the Panel is expanded
+     */
+    @Input()
+    collapseLabel = this._panelConfig.collapseLabel;
+
+    /** Output event triggered when the Expand button is clicked */
+    @Output()
+    panelExpandChange = new EventEmitter<PanelExpandChangeEvent>();
+
+    /**
+     * @hidden
+     * Button label based on the current state
+     */
+    expandAriaLabel: string;
+
+    /** @hidden */
+    @ContentChild(PanelActionsComponent)
+    panelActionsComponent: PanelActionsComponent;
+
+    /** @hidden */
+    @ContentChild(PanelContentComponent)
+    panelContentComponent: PanelContentComponent;
+
+    /** @hidden */
+    @ViewChild(PanelTitleDirective)
+    panelTitleDirective: PanelTitleDirective;
+
+    /** @hidden */
+    _contentDensity = this._panelConfig.contentDensity;
```

Comment:
Shouldn't this be labelled private since it's hidden and you are using `_` ?
Also maybe it's a good idea to add type

Reply:
I'm using `_` because `contentDensity` already used as a setter.
`_contentDensity` is not private because it's used in the component template.

Reply:
I don't use `contentDensity` getter here because we try to keep template thin and efficient that's why we avoid using any functions within it.

Reply:
Added type explicitly.